src/contract/assembled_transaction.ts

```diff
@@ -800,33 +803,40 @@ export class AssembledTransaction<T> {
      * Sign all auth entries for this account. Default: the account that
      * constructed the transaction
      */
-    publicKey?: string;
+    address?: string;
     /**
-     * You must provide this here if you did not provide one before. Default:
-     * the `signAuthEntry` function from the `Client` options. Must
-     * sign things as the given `publicKey`.
+     * You must provide this here if you did not provide one before and you are not passing `authorizeEntry`. Default: the `signAuthEntry` function from the `Client` options. Must sign things as the given `publicKey`.
```

Comment:
I'm confused: why do we have both `signAuthEntry` and `authorizeEntry`?
Also what does authorizeEntry not do that someone may want to do?

Reply:
Good question and it's the naming is what throws this all into confusion. `signAuthEntries` by default ultimately actually only signs the `xdr.HashIdPreimage` and the `js-sdk` `authorizeEntry` hard codes to the pattern that the ed25519 signer expects. But given a custom account the auth credentials signature object can be any type so we can't require all auth entries be signed by the `js-sdk` `authorizeEntry`. There will be a potentially unlimited number of ways you actually want to craft the auth entries signature. Thus the need to pass in a custom `authorizeEntry` function.

I do think renaming the `signAuthEntry` arg to `signHashIdPreimage` would help clear up this confusion.

A lot of the original `js-sdk` designs are a little bit spaghetti now because it was all built without the understanding of how `__check_auth` works. Eventually we should rewrite that but I'd rather assume the common case and just allow for bypassing it. At some point in the future we can reform the actual methods to be more flexible.

Reply:
Ah yeah, the SDK certainly was not built with custom auth in mind since, well, custom auth means custom code. Those utility functions are definitely intended to be used with standard auth. Makes sense on the distinction, thanks for explaining!